### PR TITLE
Adds dynamic image gallery to business details page, generalize add carousel element function

### PR DESCRIPTION
**Pull request by smruthibalajee**
**Description:** Added image gallery to the top of the business details page. Gallery checks blobKeys stored in the document and dynamically adds them to create carousel elements. Made a generalized createCarouselElement function in the script file that takes in a carousel id and blobKey to add an image to the specified carousel.

Screenshot 2020-08-04 at 1 49 34 PM

### CODE REVIEW

File: ripple/src/main/webapp/script.js [hunk 1]
```diff
@@ -355,6 +355,30 @@ function isWithinRange(target, startRange, endRange, lessThanAndEquals=false) {
...
+    // Append newly created card element to the container
+    var carouselElement = document.getElementById(id);
+    var carouselInner = document.getElementById("firstRow");
+    carouselInner.innerHTML = carouselInner.innerHTML + content + "\n";  
+    // Only want to append the string active once
+    if (!carouselInner.firstElementChild.className.includes("active")) {
+        carouselInner.firstElementChild.className += " active";
+    }
+    $(carouselElement).carousel({slide : true, interval : false});
+    }
```
Comment (peter14f) [inline ripple/src/main/webapp/script.js:379]: should line up with the opening bracket of the if block

File: ripple/src/main/webapp/script.js [hunk 2]
```diff
@@ -355,6 +355,30 @@ function isWithinRange(target, startRange, endRange, lessThanAndEquals=false) {
...
+        </div>
+    `;
+    console.log(content);
+    // Append newly created card element to the container
+    var carouselElement = document.getElementById(id);
+    var carouselInner = document.getElementById("firstRow");
+    carouselInner.innerHTML = carouselInner.innerHTML + content + "\n";  
+    // Only want to append the string active once
+    if (!carouselInner.firstElementChild.className.includes("active")) {
+        carouselInner.firstElementChild.className += " active";
```
Comment (peter14f) [inline ripple/src/main/webapp/script.js:376]: Remove extra indentation

File: ripple/src/main/webapp/script.js (hunk 1 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/script.js]: fixed

File: ripple/src/main/webapp/script.js (hunk 2 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/script.js]: fixed

File: ripple/src/main/webapp/businessdetails.js [hunk 3]
```diff
@@ -63,7 +63,11 @@ function bdOnload() {
   // Access 'businessDetails' collection
   var lambda = (doc) => {
         if (doc.exists) {
-          // TODO: Read galleryBlobKeys
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/businessdetails.js:66]: Add a brief comment

Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js:66]: similar to the other reads I have in this func

File: ripple/src/main/webapp/script.js [hunk 4]
```diff
@@ -355,6 +355,30 @@ function isWithinRange(target, startRange, endRange, lessThanAndEquals=false) {
...
 }
 
+/* Function that creates a generic carousel image element. Used in businessdetails and managebusiness. */
+function createCarouselElement(blobKey, id) {
+  if (blobKey != null) {
+    console.log(blobKey);
+    var content = `
+        <div class="carousel-item col-md-4">
+        <div class="card">
+            <img class="card-img-top img-fluid" id="card-dynamic-image" src="/serve?blob-key=${blobKey}"></img>
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/script.js:365]: nit: decraese spacing

File: ripple/src/main/webapp/businessdetails.js (hunk 3 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/businessdetails.js:66]: fixed

File: ripple/src/main/webapp/script.js (hunk 4 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/script.js:365]: fixed

**Pull request merged**